### DISCUSSION TITLE
What are the consequences of reopening closed entityManager?

### DISCUSSION THREAD

**Discussion by f1amy** [doctrine/doctrine2#10425] [Support Questions]
Hello, I'm trying to deal with deadlocks and wait timeouts in a PHP consumer.

I Implemented retry functionality but getting `The EntityManager is closed.` exception on second try.
We're using old Symfony and most of our services and repositories use `entityManager` directly (without proxy).
So we can't use `resetManager()` method to replace closed entity manager with a new one everywhere at once.

I come up with the following solution which seems to be working for us.
But what are the "unforeseen" consequences of doing so?

```php
    /**
     * @param EntityManager $entityManager
     * @return void
     */
    private function reopenEntityManager(EntityManager $entityManager): void
    {
        if ($entityManager->isOpen()) {
            throw new InvalidArgumentException('EntityManager is not closed to be reopened.');
        }

        $this->logger->notice('Reopening closed entityManager.');

        $reflection = new ReflectionObject($entityManager);
        $property = $reflection->getProperty('closed');
        $property->setAccessible(true);
        $property->setValue($entityManager, false);

        $entityManager->clear();
        $entityManager->getConnection()->close();
    }
```

**Comment by mpdude** [comment]
You should try to figure out (I don’t know) what it means that an EM has been closed, and what the reasons were to design the API in a way that it cannot easily be reopened.

Do yourself and future team mates a favor and try not to hack the EM as shown above.
